langserver/completion.go

```diff
@@ -452,3 +454,23 @@ func getEditRange(location *cache.Location, oldname string) (editRange protocol.
 
 	return
 }
+
+// getMatches returns fuzzy matches for a slice of string and a pattern.
+func getMatches(pattern string, names []string) fuzzy.Matches {
+	if pattern == "" {
+		var (
+			match   fuzzy.Match
+			matches fuzzy.Matches
+		)
+		sort.Strings(names)
+		match.MatchedIndexes = append(match.MatchedIndexes, 0) // match the first rune for all
+		for i, name := range names {
+			match.Str = name
+			match.Index = i
+			matches = append(matches, match)
+			match.Score = len(names) - i
```

Comment:
```suggestion
			matches = append(matches, fuzzy.Match{
	                Str: name,
	                Index: i,
	                Score:  len(names) - i,
		       })
```

Reply:
Also the results should be deduplicated.

Reply:
how do you mean? appending to *completion directly?

Reply:
Nevermind, this comment was based on the assumption, that fuzzy deduplicates results, which is [wrong](https://play.golang.org/p/g_SYiDUwK9O).

Reply:
But using the first suggestion IMHO still makes the code more readable.

Matched indexes should be kept as `nil` since there are no matched indexes in that case.

Reply:
updated